src/or/proto_socks.c

```diff
@@ -250,6 +256,8 @@ parse_socks5_methods_request(const uint8_t *raw_data, socks_request_t *req,
     goto end;
   }
 
+  *drain_out = (size_t)parsed;
```

Comment:
Better!  But let's assert that parsed >= 0 in this case.

Reply:
(And let's make sure drain_out is initialized to 0 in this function too)

Reply:
Fixed both in bfa4a7da261b4acb75878522d70c305378c1753e.